external-crates/move/crates/move-stackless-bytecode/src/spec_purity_analysis.rs

```diff
@@ -57,64 +81,26 @@ impl SpecPurityAnalysis {
                             let module_name = env.symbol_pool().string(module.get_name().name());
 
-                            if module_name.as_str() == GlobalEnv::PROVER_MODULE_NAME {
+                            if module_name.as_str() == GlobalEnv::PROVER_MODULE_NAME || module_name.as_str() == GlobalEnv::SPEC_MODULE_NAME {
                                 continue;
                             }
 
                             if RESTRICTED_MODULES.contains(&module_name.as_str()) {
                                 results.insert(builder.get_loc(*attr)); 
                             }
 
-                            let modif_function = self.find_modifiable_locations_in_bytecode_internal(&module.get_function(*func_id), env, skip, &mut visited_funcs);
-                            if modif_function {
-                                results.insert(builder.get_loc(*attr)); 
-                            }
-                        },
-                        _ => {}
-                    };
-                },
-                _ => {},
-            }
-        }
-        results
-    }
-
-    pub fn find_modifiable_locations_in_bytecode_internal(&self, func_env: &FunctionEnv, env: &GlobalEnv, skip: &Option<Operation>, visited_funcs: &mut BTreeSet<FunId>) -> bool {
-        let internal_generator: StacklessBytecodeGenerator<'_> = StacklessBytecodeGenerator::new(&func_env);
-        let binding = internal_generator.generate_function();
-        let internal_func_target = FunctionTarget::new(&func_env, &binding);
-        let internal_code = internal_func_target.get_bytecode();
-
-        for param in func_env.get_parameters() {
-            if param.1.is_mutable_reference() {
-                return true;
-            }
-        }
-
-        for cp in internal_code {
-            match cp {
-                Bytecode::Call(_, _, operation, _, _) => {
-                    match operation {
-                        Operation::Function(mod_id,func_id, _) => {
-                            if !visited_funcs.insert(func_id.clone()) {
-                                continue;
-                            }
-
-
-                            let module = env.get_module(*mod_id); 
-                            let module_name = env.symbol_pool().string(module.get_name().name());
-
-                            if module_name.as_str() == GlobalEnv::PROVER_MODULE_NAME {
+                            let internal_data = targets.get_data(&mod_id.qualified(*func_id), &FunctionVariant::Baseline);
+                            if internal_data.is_none() {
                                 continue;
                             }
+                                
+                            let annotation = internal_data
+                                .unwrap()
+                                .annotations.get::<PurityVerificationInfo>()
+                                .expect("Function expect to be already scanned");
```

Comment:
I initially said not to worry about recursive functions, but now I think we should handle that properly. I found this example in the codebase: https://github.com/asymptotic-code/sui/blob/next/external-crates/move/crates/move-stackless-bytecode/src/borrow_analysis.rs#L391-L420